libraries/AP_HAL_ChibiOS/hwdef/Aocoda-RC-H743Dual/hwdef.dat

```diff
@@ -0,0 +1,218 @@
+# hw definition file for processing by chibios_pins.py
+# for Aocoda-RC-H743Dual: MPU6000/BMI270x2
+
+# MCU class and specific type
+MCU STM32H7xx STM32H743xx
+
+# board ID for firmware load
+APJ_BOARD_ID AP_HW_AOCODA-RC-H743DUAL
+
+# crystal frequency, setup to use external oscillator
+OSCILLATOR_HZ 8000000
+
+# flash size
+FLASH_SIZE_KB 2048
+
+env OPTIMIZE -Os
+
+# bootloader takes first sector
+FLASH_RESERVE_START_KB 128
+
+# ChibiOS system timer
+STM32_ST_USE_TIMER 12
+define CH_CFG_ST_RESOLUTION 16
+
+# USB setup
+USB_STRING_MANUFACTURER "Aocoda-RC-H743Dual"
+
+# Now we define the pins that USB is connected on.
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# OTG USB Detect
+PE2 EXT_CS2 CS
+
+# These are the pins for SWD debugging with a STlinkv2 or black-magic probe.
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 for IMU1 (BMI270/MPU6000)
+PC15 IMU1_CS CS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PD7 SPI1_MOSI SPI1
+
+# SPI2 for MAX7456 OSD
+PB12 MAX7456_CS CS
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# SPI3 for SPI flash - W25N01GV
+PD3 FLASH_CS CS
+PB3 SPI3_SCK SPI3
+PB4 SPI3_MISO SPI3
+PB5 SPI3_MOSI SPI3
+
+# SPI4 for IMU2 (BMI270)
+PE11 IMU2_CS CS
+PE12 SPI4_SCK SPI4
+PE13 SPI4_MISO SPI4
+PE14 SPI4_MOSI SPI4
+
+# two I2C bus, order of I2C buses
+I2C_ORDER I2C2 I2C1
+
+# I2C1
+PB6 I2C1_SCL I2C1
+PB7 I2C1_SDA I2C1
+
+# I2C2
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# ADC for Power
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 11.0
+define HAL_BATT_CURR_SCALE 40.0
+
+# ADC for airspeed
+PC4 PRESSURE_SENS ADC1 SCALE(2)
+define HAL_DEFAULT_AIRSPEED_PIN 4
+
+# ADC for rssi
+PC5 RSSI_ADC ADC1
+define BOARD_RSSI_ANA_PIN 8
+
+# Define LEDs
+# blue LED0 marked as ACT
+# green LED1 marked as B/E
+PE3 LED0 OUTPUT LOW GPIO(90) # blue
+PE4 LED1 OUTPUT LOW GPIO(91) # orange
+define HAL_GPIO_A_LED_PIN 91
+define HAL_GPIO_B_LED_PIN 90
+define HAL_GPIO_LED_OFF 1
+
+# The normal usage of this ordering is:
+# 1) SERIAL0: console (primary mavlink, usually USB)
+# 2) SERIAL1: RC input
+# 3) SERIAL2: GPS
+# 4) SERIAL3: VTX
+# 5) SERIAL4: Telemetry
+# 6) SERIAL5: not available
+# 7) SERIAL6: ESC Telemetry
+# 8) SERIAL7: spare
+# 9) SERIAL8: Optical Flow
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 EMPTY USART6 UART7 UART8
+
+# USART1 (RC input), SERIAL1
+PA9  USART1_TX USART1
+PA10 TIM1_CH3 TIM1 RCININT PULLDOWN LOW
+
+# as an alternative config setup the RX1 pin as a uart. This allows
+# for bi-directional UART based receiver protocols such as FPort
+# without any extra hardware
+PA10 USART1_RX USART1 ALT(1)

```

Comment:
I recommend you don't do it this way but use RCIN protocol instead unless you really want to support PPM. Its extra steps for users who want to use protocols using two wires - e.g. CRSF or ELRS of which there are many.

Reply:
I agree, losing PPM is not a big deal these days and it would fix your LED issue

Reply:
We orignally thought that DJI user (DJI remote control) will try to use this flight control. If FC doesn't support SBUS, then they have to buy another Remote contrl.

Anyway, if there is no other way to work around, then there should be a compromise (LED strip fixed).
```
# USART1 (RC input), SERIAL1
PA9  USART1_TX USART1
PA10 USART1_RX USART1
```

Can we leave a message on wiki, if someone DO need to support sbus receiver, just contact us. Then we can share info with them?

Reply:
SBUS will work no problem - its an H743 so we can do pin inversion in software, and indeed do on that chip.

Reply:
Rx1 is wired to DJI connector. I'm NOT sure how can we make subs and CRSF work with a config option?

Reply:
Not sure what you think the issue is. The only issue I can imagine is that TX1 is not actually pinned out. What is subs?

Reply:
I hope Rx1 can be configured as SBUS (for DJI) or CRSF (for ELRS), and at the mean time LED stripe can work properly.

> What is subs?
subs should be SBUS. typo issue.

Reply:
right so you don't need to use RCININT for that to happen. Just use SerialProtocol_RCIN and then all will work

Reply:
Thanks for confirmation.